node/src/api/wallet/handlers.rs

```diff
@@ -0,0 +1,177 @@
+use std::convert::{Infallible, TryFrom};
+
+use crate::{
+    api::{
+        response::{error, Response, ResponseError, ResponseResult},
+        types,
+        types::Cursor,
+        wallet::{requests, responses},
+    },
+    wallet::{BuiltTx, TxAction, Wallet},
+    wallet_manager::WalletRef,
+};
+use accounts::{Address, AddressLabel};
+use curve25519_dalek::scalar::Scalar;
+use keytree::{Xprv, Xpub};
+
+/// Creates a new wallet
+pub(super) async fn new(
+    request: requests::NewWallet,
+    wallet: WalletRef,
+) -> ResponseResult<responses::NewWallet> {
+    let requests::NewWallet { xpub, label } = request;
+    let mut wallet_ref = wallet.write().await;
+    if wallet_ref.wallet_exists() {
+        if let Err(_) = wallet_ref.clear_wallet() {
+            return Err(error::cannot_delete_file());
+        }
+    }
+    let label = AddressLabel::new(label).ok_or_else(|| error::invalid_address_label())?;
+    let xpub = Xpub::from_bytes(&xpub).ok_or_else(|| error::invalid_xpub())?;
+
+    let new_wallet = Wallet::new(label, xpub);
+    wallet_ref
+        .initialize_wallet(new_wallet)
+        .expect("We previously deleted wallet, there are no other errors when initializing wallet");
+
+    Ok(responses::NewWallet)
+}
+
+/// Returns wallet's balance.
+pub(super) async fn balance(wallet: WalletRef) -> ResponseResult<responses::Balance> {
+    let mut wallet_ref = wallet.read().await;
+    let wallet = wallet_ref
+        .wallet_ref()
+        .map_err(|_| error::wallet_does_not_exist())?;
+
+    let balances = wallet
+        .balances()
+        .map(|balance| (balance.flavor.to_bytes(), balance.total))
+        .collect::<Vec<_>>();
+
+    Ok(responses::Balance { balances })
+}
+
+/// Lists annotated transactions.
+pub(super) async fn txs(
+    _cursor: Cursor,
+    _wallet: WalletRef,
+) -> ResponseResult<responses::WalletTxs> {
+    unimplemented!()
+}
+
+/// Generates a new address.
+pub(super) async fn address(wallet: WalletRef) -> ResponseResult<responses::NewAddress> {
+    let mut wallet_ref = wallet.write().await;
+    let update_wallet = |wallet: &mut Wallet| Ok(wallet.create_address());
+
+    match wallet_ref.update_wallet(update_wallet) {
+        Ok(address) => Ok(responses::NewAddress {
+            address: address.to_string(),
+        }),
+        Err(crate::Error::WalletNotInitialized) => Err(error::wallet_does_not_exist()),
+        _ => Err(error::wallet_updating_error()),
+    }
+}
+
+/// Generates a new receiver.
+pub(super) async fn receiver(
+    req: requests::NewReceiver,
+    wallet: WalletRef,
+) -> ResponseResult<responses::NewReceiver> {
+    let mut wallet_ref = wallet.write().await;
+    let update_wallet = |wallet: &mut Wallet| {
+        let requests::NewReceiver { flv, qty, exp } = req; // TODO: expiration time?
```

Comment:
Hmm.. I'm starting to think that the expiration time is the storage policy of the node. For about we remove this parameter from the request entirely for now?